pkgs/development/libraries/modern-cpp-kafka/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv, fetchFromGitHub, cmake, boost, gtest, rapidjson, rdkafka }:
+
+stdenv.mkDerivation rec {
+  pname = "modern-cpp-kafka";
+  version = "2023.03.07";
+
+  src = fetchFromGitHub {
+    owner = "morganstanley";
+    repo = "modern-cpp-kafka";
+    rev = "v${version}";
+    sha256 = "sha256-7hkwM1YbveQpDRqwMZ3MXM88LTwlAT7uB8NL0t409To=";
+  };
+
+  patches = [
+    # The lib expects headers and lib is in common dir, but nix splits then into out and dev derivations
+    ./gtest_envs.patch
+  ];
+  # Envs used by the patch
+  GTEST_INCLUDE_DIR = "${gtest.dev}/include";
+  GTEST_LIBRARY_DIR = "${gtest}/lib";
+
+  cmakeFlags = [
+    "-DCMAKE_INCLUDE_PATH=${rdkafka}/include/librdkafka;${rapidjson}/include/rapidjson"
+  ];
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ boost.dev gtest gtest.dev rapidjson rdkafka ];
```

Comment:
Oh, also i think gtest could be a nativeBuildInput, also?

Reply:
This lib is header-only, so I'm not sure which dependencies go where. The build is mostly test-only. "Production build" is to copy headers to out

Reply:
Hm. So maybe the rdkafka and rapidjson deps should be propagated? On that I'm not so clear...

Reply:
I am pretty clear that the `test-only` deps should be `nativeBuildInputs` though.

Reply:
I've moved `gtest` and `rapidjson` to native deps, since both are test only.
Now only boost and rdkafka remains in buildInputs.